feature-flags-tmp/feature_flags/feature_flags.py

```diff
@@ -0,0 +1,222 @@
+        if re.search(r"[^a-zA-Z0-9_]+", self.new_flag_modal_flag_name):
+            self.new_flag_modal_error = "Flag name must be contain only one or more alphanumeric or _ chars"
+            return
+        if self.new_flag_modal_flag_name in self.latest_feature_flags_view():
+            self.new_flag_modal_error = "Flag already exists"
+            return
+        self.pending_creates_or_updates[self.new_flag_modal_flag_name] = self.new_flag_modal_flag_value
+        self.reset_new_flag_modal_state()
+
+    def new_flag_modal_cancel(self):
+        self.reset_new_flag_modal_state()
+
+    def stage_delete_feature_flag(self, k: str):
+        self.pending_deletes.add(k)
+
+
+def index() -> rx.Component:
+    return rx.fragment(
+        rx.color_mode_button(rx.color_mode_icon(), float="right"),
+        rx.vstack(
+            rx.heading("Flex-Flags", font_size="2em"),
+            rx.hstack(
+                rx.button("Save", on_click=FeatureFlagsState.save_to_db, color=FeatureFlagsState.save_button_color),
+                rx.button("Add new", on_click=lambda: FeatureFlagsState.set_new_flag_modal_is_open(True)),
+            ),
+            rx.cond(FeatureFlagsState.pending_deletes,
+                    rx.text(f"Pending deletes: {FeatureFlagsState.pending_deletes}", color="red")),
+            rx.cond(FeatureFlagsState.pending_creates,
+                    rx.text(f"Pending creates: {FeatureFlagsState.pending_creates}", color="red")),
+            rx.cond(FeatureFlagsState.pending_updates,
+                    rx.text(f"Pending updates: {FeatureFlagsState.pending_updates}", color="red")),
+            rx.modal(
+                rx.modal_overlay(
+                    rx.modal_content(
+                        rx.modal_header("Add new feature flag"),
+                        rx.modal_body(
+                            "Add a new feature flag",
+                            rx.hstack(
+                                rx.input(value=FeatureFlagsState.new_flag_modal_flag_name, width='30%',
+                                         on_change=FeatureFlagsState.set_new_flag_modal_flag_name),
+                                rx.input(value=FeatureFlagsState.new_flag_modal_flag_value,
+                                         on_change=FeatureFlagsState.set_new_flag_modal_flag_value,
+                                         width='70%')),
+                            rx.cond(FeatureFlagsState.new_flag_modal_error,
+                                    rx.text(FeatureFlagsState.new_flag_modal_error, color="red"))
+                        ),
+                        rx.modal_footer(
+                            rx.button(
+                                "Stage", on_click=FeatureFlagsState.new_flag_modal_stage,
+                            ),
+                            rx.button(
+                                "Cancel", on_click=FeatureFlagsState.new_flag_modal_cancel,
+                            )
+                        ),
```

Comment:
possibly a bit nicer to use an `rx.form` here as then you get `<enter>` key functionality automatically and don't need to use controlled inputs